tests/serialize/runstate/dynamodb_state_store_test.py

```diff
@@ -115,221 +57,135 @@ def small_object():
-                small_object,
-            ),
-        ]
-        store.save(key_value_pairs)
-        store._consume_save_queue()
-
-        delete = [
-            (
-                store.build_key("job_state", "two"),
-                None,
-            ),
-        ]
-        store.save(delete)
-        store._consume_save_queue()
+    @pytest.mark.parametrize(
+        "object_fixture_name",
+        ["small_jobrun_state_object", "large_jobrun_state_object"],
+    )
+    def test_save_and_restore(self, store, object_fixture_name, request):
+        """Verify that objects of different sizes can be saved and restored correctly."""
+        state_object = request.getfixturevalue(object_fixture_name)
+        key = store.build_key("job_run_state", "test_jobrun")
+        key_value_pairs = [(key, state_object)]
 
-        assert store.save_errors == 0
-        # Try to restore both, we should just get one back
-        keys = [
-            store.build_key("job_state", "two"),
-            store.build_key("job_run_state", "four"),
-        ]
-        with mock.patch("tron.config.static_config.load_yaml_file", autospec=True), mock.patch(
-            "tron.config.static_config.build_configuration_watcher", autospec=True
-        ):
-            vals = store.restore(keys)
-        assert vals == {keys[1]: small_object}
-
-    def test_save_more_than_4KB(self, store, small_object, large_object):
-        key_value_pairs = [
-            (
-                store.build_key("job_state", "two"),
-                large_object,
-            ),
-        ]
         store.save(key_value_pairs)
         store._consume_save_queue()
 
+        restored_vals = store.restore([key])
         assert store.save_errors == 0
-        keys = [store.build_key("job_state", "two")]
-
-        with mock.patch("tron.config.static_config.load_yaml_file", autospec=True), mock.patch(
-            "tron.config.static_config.build_configuration_watcher", autospec=True
-        ):
-            vals = store.restore(keys)
-        for key, value in key_value_pairs:
-            assert_equal(vals[key], value)
-
-    @pytest.mark.parametrize("read_json", [False, True])
-    def test_restore_more_than_4KB(self, store, small_object, large_object, read_json):
-        keys = [store.build_key("job_state", i) for i in range(3)]
-        value = large_object
-        pairs = zip(keys, (value for i in range(len(keys))))
-        store.save(pairs)
+        assert restored_vals[key] == state_object
+
+        num_partitions = store._get_num_of_partitions(key)
+        if object_fixture_name == "small_jobrun_state_object":
+            assert num_partitions == 1
+        else:
+            # For large objects, we expect multiple partitions, the number of partitions can vary depending on our object size
+            # and the DynamoDB partition size limit. We assert that there is more than one partition.
+            assert num_partitions > 1
```

Comment:
we should try to avoid conditionals in tests - that said, not sure how to handle this specific case without having one of the parameterized values to pass through be a lambda or something :p